### PR TITLE
WDPT BER Patrick

### PR DESCRIPTION
**Pull request by Patrick-Groot**

### CODE REVIEW

File: src/movies.js [hunk 1]
```diff
@@ -1,25 +1,93 @@
...
 // BONUS - Iteration 8: Best yearly score average - Best yearly score average
-function bestYearAvg(moviesArray) {}
+/* function bestYearAvg(moviesArray) {
+    if (moviesArray.length === 0){
+        return null;
+    }
+    const orderedByYearArray = orderByYear(moviesArray);
+    let moviesPerYear = 0;
+    let scorePerYear = 0;
+    orderedByYearArray.reduce((a,b) => {
```
[the remark this answers is not in the document]
Comment (trebeljahr) [reply src/movies.js:87]: possible solution for this if you want to pick it apart 
```js
function bestYearAvg(moviesArray) {
  if (!moviesArray.length) return null;

  const moviesByYear = moviesArray.reduce((groups, movie) => {
    if (!groups[movie.year]) {
      return { ...groups, [movie.year]: [movie] };
    }
    groups[movie.year].push(movie);
    return groups;
  }, {});

  const bestYear = Object.entries(moviesByYear).reduce(
    (best, [year, moviesForThatYear]) => {
      const score = scoresAverage(moviesForThatYear);
      if (score >= best.score) {
        const tiedAndCurrentIsOlder = best.score === score && best.year < year;
        return tiedAndCurrentIsOlder ? best : { year, score: score };
      }

      return best;
    },
    { score: -Infinity, year: Infinity }
  );

  return `The best year was ${bestYear.year} with an average score of ${bestYear.score}`;
}
```

**Review by trebeljahr — COMMENTED** [review]
really good work, though you have this one persistent error where you index into an array to get a value that you already had access to from the array methods arguments anyways.

**Comment by Patrick-Groot** [comment]
> really good work, though you have this one persistent error where you index into an array to get a value that you already had access to from the array methods arguments anyways.

Thank you so much! Yeah, I was a bit confused about that at the time, what exactly I am accessing - and I tried it like this at first and it worked, so I just kept doing it like that :sweat_smile:

File: src/movies.js [hunk 2]
```diff
@@ -1,25 +1,93 @@
...
+function scoresAverage(moviesArray) {
+    if (moviesArray.length === 0){
+        return 0;
+    }
+    const averageRating = moviesArray.reduce((agg, currentMovie) => {
+        if ("score" in currentMovie === false || currentMovie.score === "") {
+            currentMovie.score = 0;
+        }
+        return agg + currentMovie.score}, 0) / moviesArray.length;
+    return Math.round(averageRating * 100) / 100;
```
[the remark this answers is not in the document]
Comment (Patrick-Groot) [reply src/movies.js:34]: I didn't really test it a lot for exceptions. Just found it on Google and liked doing it this way :)

**Pull request closed without merging**